src/components/AgentRow.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { useContext } from 'react';
+import { Checkbox, Tooltip } from 'antd';
+import { UIDisplayEntry } from '@aics/simularium-viewer/type-declarations/simularium/SelectionInterface';
+
+import { TOOLTIP_COLOR, CheckboxState } from '../constants';
+import { VisibilityContext } from '../AgentVisibilityContext';
+
+interface AgentRowProps {
+  agent: UIDisplayEntry;
+}
+
+const AgentRow: React.FC<AgentRowProps> = (
+  props: AgentRowProps
+): JSX.Element => {
+  const { agent } = props;
+
+  const { handleAgentCheckboxChange, hiddenAgents } =
+    useContext(VisibilityContext);
+
+  const getCheckboxState = () => {
+    if (hiddenAgents[agent.name]?.length === 0) {
+      return CheckboxState.Unchecked;
+    }
+    return CheckboxState.Checked;
+  };
```

Comment:
I'm assuming indeterminate states will get added later?

Reply:
Yes, I'm trying to keep the future out of these PRs but there are some cases where it just makes more sense to anticipate what's coming.